code/game/machinery/firealarm.dm

```diff
@@ -114,12 +129,29 @@
 	var/area/A = get_area(src)
 	A.firealert(src)
 	playsound(src.loc, 'goon/sound/machinery/FireAlarm.ogg', 75)
+	if(timerid)
+		return
+	flash()
+
+/obj/machinery/firealarm/proc/flash()
+	if (!powered())
+		return
+	if(!is_operational())
+		return
+
+	flash_lighting_fx(FLASH_LIGHT_RANGE, flash_power, flash_color)
+	if(!timerid)
+		timerid = addtimer(CALLBACK(src, .proc/flash, world.time), 30, TIMER_UNIQUE|TIMER_STOPPABLE|TIMER_LOOP)

```

Comment:
could probably use TIMER_OVERRIDE too, but don't think you'd have to with the way you did it.

Reply:
Mayhaps. I just nabbed this from sound loop code because it was the first thing that came to mind for a 'looping' timer.